pypesto/hierarchical/inner_calculator_collector.py

```diff
@@ -102,14 +101,17 @@ def __init__(
         self.inner_calculators: list[
             AmiciCalculator
         ] = []  # TODO make into a dictionary (future PR, together with .hierarchical of Problem)
+
+        self.semiquant_observable_ids = None
+        self.relative_observable_ids = None
+
         self.construct_inner_calculators(
             petab_problem, model, edatas, inner_options
         )
 
         self.quantitative_data_mask = self._get_quantitative_data_mask(edatas)
 
         self._known_least_squares_safe = False
-        self.semiquant_observable_ids = None
```

Comment:
Bugfix.
`semiquant_observable_ids ` would be set to the right value in `construct_inner_calculators` and then was always overwritten to None in line 112.